extensions/resource-deployment/src/ui/radioGroupLoadingComponentBuilder.ts

```diff
@@ -45,7 +45,14 @@ export class RadioGroupLoadingComponentBuilder implements azdata.ComponentBuilde
 			}
 
 			let options: (string[] | azdata.CategoryValue[]) = optionsInfo.values!;
-			let defaultValue: string = optionsInfo.defaultValue!;
+			let defaultValue: string | ResourceTypeOptionValue = optionsInfo.defaultValue!;
+			if (optionsInfo.defaultValue) {
+				if ((<ResourceTypeOptionValue>optionsInfo.defaultValue).displayName) {
```

Comment:
I'd suggest switching the check around and checking the type instead. That should remove the need to cast the value again in the if statement since it can use the type narrowing to know what type it is. 
```
if (typeof optionsInfo.defaultValue === 'string') { 
...
} else {
...
}
```